target/riscv/op_helper_cheri.c

```diff
@@ -156,7 +156,11 @@ void HELPER(cspecialrw)(CPUArchState *env, uint32_t cd, uint32_t cs,
 {
     uintptr_t _host_return_address = GETPC();
     // Ensure that env->PCC.cursor is correct:
-    cpu_restore_state(env_cpu(env), _host_return_address, false);
+    /*
+     * This breaks -icount because it triggers early deallocation of the TB.
+     * It seems that it is only required for logging anyway.
+     */
+    /* cpu_restore_state(env_cpu(env), _host_return_address, false); */
```

Comment:
I believe this is needed to read PCC correctly - we need a cursor that is up to date. But maybe the PC is already saved, might be possible to just assert that pc is current instead of calling this function?

Reply:
I re-enabled this and I can't seem to be able to reproduce the -icount bug. I suspect one of the latest merges with upstream qemu changed the code that invalidates the TB. I am hitting a loop condition that seems to trigger an interrupt loop in CheriBSD with `-icount auto` though. This is an unrelated problem, but I'm not sure how we can fix it.